students/njschafi/Lesson02/assignment/charges_calc.py

```diff
@@ -0,0 +1,135 @@
+        console_handler = logging.StreamHandler()
+        console_handler.setFormatter(formatter)
+        logger.setLevel(logging.DEBUG)
+        if level == 3:
+            # Debug: Should be shown on screen only(never saved to logfile).
+            file_handler.setLevel(logging.WARNING)
+            console_handler.setLevel(log_levels[level])
+        else:
+            logger.setLevel(log_levels[level])
+        logger.addHandler(file_handler)
+        logger.addHandler(console_handler)
+
+def parse_cmd_arguments():
+    '''
+    Parse command line arguments
+    '''
+    parser = argparse.ArgumentParser(description='Process some integers.')
+    parser.add_argument('-i', '--input', help='input JSON file', required=True)
+    parser.add_argument('-o', '--output', help='ouput JSON file', required=True)
+    parser.add_argument('-d', '--debug', help='debug level',\
+                        required=False, default=0)
+    return parser.parse_args()
+
+def load_rentals_file(filename):
+    '''
+    Load input file
+    '''
+    try:
+        with open(filename) as file: # try inputting a none json file
+            data = json.load(file)
+    except FileNotFoundError as ex:
+        logging.error('Can not load file: %(filename)s. Either because it is'
+                      ' not in JSON format or'
+                      ' incorrectly typed.')
+        logging.warning('No data will be passed which may'
+                        ' result in empty output file.')
+        logging.debug('file: %(filename)s is not formatted correctly.')
+        logging.debug(ex)
+        data = None
+    return data
+
+def calculate_additional_fields(data):
+    '''
+    Calculate additional fields based on source.json file
+    '''
+    if data is None:
+        logging.debug('No data passed in. File is either incorrect or empty.')
+        logging.warning('Data inputted is None. Check file input.')
+        return None
+
+    for value in data.values():
+        try:
+            rental_start =\
+            datetime.datetime.strptime(value['rental_start'], '%m/%d/%y')
+            rental_end =\
+            datetime.datetime.strptime(value['rental_end'], '%m/%d/%y')
+        except ValueError as err:
+            logging.warning('Error: %s. Product: %s',
+                            err, value['product_code'])
+            logging.debug(value)
+
+        value['total_days'] = (rental_end - rental_start).days
+        if value['total_days'] < 0:
+            logging.warning('Negative amount of rental days. Product: %s',
+                            value['product_code'])
+            logging.debug('Amount of Days: %s', value['total_days'])
+
+        try:
+            value['total_price'] = value['total_days'] * value['price_per_day']
+            # can't take sqrt of negative number
+            value['sqrt_total_price'] = math.sqrt(value['total_price'])
+        except ValueError as err:
+            # logging.error('total_price is negative:'
```

Comment:
This won't be flagged by Pylint as an issue, nevertheless, always avoid leaving commented code in a release (or a submission, in this case). The number one risk is that someone uncomments it by error later on, and it creates unnecessary clutter in what is otherwise very high quality coding.